### PR TITLE
[runtime] Fix build by building sgen first, rather than invoking out of order

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

**Comment by vargaz** [comment]
LGTM

**Comment by lewurm** [comment]
@monojenkins merge

File: mono/metadata/Makefile.am [hunk 1]
```diff
@@ -313,16 +315,14 @@ if SHARED_MONO
 pedump_SOURCES =		\
 	pedump.c
 
-$(top_srcdir)/mono/sgen/libmonosgen.la:
-	make -w -C $(top_srcdir)/mono/sgen libmonosgen.la
-
-pedump_LDADD = $(sgen_libraries) $(top_srcdir)/mono/sgen/libmonosgen.la ../io-layer/libwapi.la ../utils/libmonoutils.la \
-	$(GLIB_LIBS) -lm $(LIBICONV) $(PEDUMP_DTRACE_OBJECT)
+pedump_LDADD = $(sgen_libraries) ../sgen/libmonosgen.la \
```
Comment (alexrp) [inline mono/metadata/Makefile.am]: Why relative paths? I doubt this will work in out-of-tree builds.

File: mono/Makefile.am [hunk 2]
```diff
@@ -30,7 +30,7 @@ monotouch-do-clean:
 	  (cd $$subdir && $(MAKE) $(AM_MAKEFLAGS) $$target); \
     done;
 else
-SUBDIRS = arch utils io-layer cil metadata $(sgen_dirs) mini dis tests unit-tests benchmark profiler
+SUBDIRS = arch utils io-layer cil $(sgen_dirs) metadata mini dis tests unit-tests benchmark profiler
```
Comment (akoeplinger) [inline mono/Makefile.am]: you need to do the same changes in L6 and L9.

**Comment by akoeplinger** [comment]
what about @vargaz's question from https://github.com/mono/mono/pull/3066#issuecomment-224507163:
> How about we disable pedump and/or move it to tools/? Is anybody really using it ?

Moving to tools/ seems like a solution as the products don't need it, but we need to keep it in desktop Mono.

**Pull request closed without merging**